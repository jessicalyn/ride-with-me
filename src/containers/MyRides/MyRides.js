import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Query } from 'react-apollo';
import { gql } from 'apollo-boost';
import { Link, Redirect } from 'react-router-dom';
import { Loader } from '../../components/Loader/Loader';

export class MyRides extends Component {
  
  render() {
    const userUuid = this.props.user.uuid
    const MY_RIDES = gql`
      query ($userUuid: String!){
        myRides(userUuid: $userUuid) {
          id
          description
          price
          availableSeats
          departureDate
          status
          driver { id uuid firstName lastName }
          ridepassengerSet {
            passenger { id firstName lastName }
          }
          endCity { id name }
          startCity { id name }
        }
      }
    `
    return(
      <div className="my-rides-container">
        { !this.props.user && <Redirect to='/login' /> }
        <h2>My Rides</h2>
        <Query query={MY_RIDES} variables={{ userUuid }} >
          {({ loading, error, data }) => {
            if (loading) return <Loader />
            if (error) return <p>{ error }</p>
            return data.myRides.map(({ id, departureDate, driver, endCity, startCity }) => (
              <div key={id} className="ride-container">
                <div>
                  <h4>Ride with { driver.firstName }</h4>
                  <p>{ startCity.name } to { endCity.name }</p>
                  <p>Departure Date: { departureDate }</p>
                  <Link to={`/rides/${id}`} myrides={ data.myRides } key={id}>See Ride Info</Link>
                </div>
                <div>
                  <img src={`https://maps.googleapis.com/maps/api/staticmap?size=300x200&markers=color:green%7Clabel:1%7C${startCity.name}&markers=color:red|label:2|${endCity.name}&key=AIzaSyC1l-GGLlVWnXFaybX2DaJr4bS4ANmAWMo`} alt="map" />
                </div>
              </div>
              ))
          }}
        </Query>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(MyRides)