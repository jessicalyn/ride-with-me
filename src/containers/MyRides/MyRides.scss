.my-rides-container {
  box-shadow: $box-shadow;
  width: 80%;
  margin: 20px auto;
  background-color: white;
  border-radius: 10px;
  padding: 10px;
}

.ride-container {
  border: 1px solid grey;
  margin: 30px;
  border-radius: 8px;
  padding: 10px;
  box-shadow: 0px 2px 5px 1px grey;
  display: flex;
  justify-content: space-around;
  align-items: center
}