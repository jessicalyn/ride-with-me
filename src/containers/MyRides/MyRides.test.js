import React from 'react';
import { shallow } from 'enzyme';
import { MyRides } from './MyRides';

describe('MyRides', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<MyRides />)
  })

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })
})