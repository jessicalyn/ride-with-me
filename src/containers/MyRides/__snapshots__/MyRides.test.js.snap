// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRides should match the snapshot with all data passed in 1`] = `ShallowWrapper {}`;
